tools/serverpod_cli/lib/src/util/locate_modules.dart

```
@@ -3,83 +3,72 @@ import 'dart:io';
-          ),
-        );
-      } catch (e) {
+  for (var packageInfo in packageConfig.packages) {
+    try {
+      var packageName = packageInfo.name;
+
+      if (!packageName.endsWith(_serverSuffix) && packageName != 'serverpod') {
+        continue;
+      }
+      var moduleName = moduleNameFromServerPackageName(packageName);
+
+      var packageSrcRoot = packageInfo.packageUriRoot;
+      var moduleProjectRoot = List<String>.from(packageSrcRoot.pathSegments)
+        ..removeLast()
+        ..removeLast();
+      var generatorConfigSegments = path
+          .joinAll([...moduleProjectRoot, 'config', 'generator.yaml']).split(
+              path.separator);
+
+      var generatorConfigUri = packageSrcRoot.replace(
+        pathSegments: generatorConfigSegments,
+      );
+
+      var generatorConfigFile = File.fromUri(generatorConfigUri);
+      if (!await generatorConfigFile.exists()) {
         continue;
       }
-    }
 
-    return modules;
-  } else {
-    log.error(
-      'Failed to read your server\'s package configuration. Have you run '
-      '`dart pub get` in your server directory?',
-    );
-    return null;
+      var moduleProjectUri = packageSrcRoot.replace(
+        pathSegments: moduleProjectRoot,
+      );
+
+      var migrationVersions = findAllMigrationVersionsSync(
+        directory: Directory.fromUri(moduleProjectUri),
+        moduleName: moduleName,
+      );
+
+      var moduleInfo = loadConfigFile(generatorConfigFile);
+
+      var manualNickname = manualModules[moduleName];
+      var nickname = manualNickname ?? moduleInfo['nickname'] ?? moduleName;
+
+      modules.add(
+        ModuleConfig(
+          type: GeneratorConfig.getPackageType(moduleInfo),
+          name: moduleName,
+          nickname: nickname,
+          migrationVersions: migrationVersions,
+          serverPackageDirectoryPathParts: moduleProjectRoot,
+        ),
+      );
+    } catch (e) {
+      continue;
```

Question: Is this what we want to do, silently ignore packages we cannot find on disk?

---

Yes, becuase the inspected packages could today be any package with a `_server` prefix. Once we have a more narrow definition of the projects inspected we can put higher demands here.